pkg/liquidity-mining/contracts/gauges/StakelessGaugeController.sol

```diff
@@ -0,0 +1,223 @@
+        (uint256 singleArbGaugeBridgeETH, uint256 totalArbBridgeETH) = _getArbitrumBridgeCosts();
+
+        // solhint-disable-next-line not-rely-on-time
+        uint256 currentPeriod = _roundDownTimestamp(block.timestamp);
+
+        _checkpointGauges(IGaugeAdder.GaugeType.Polygon, minRelativeWeight, currentPeriod, 0);
+        _checkpointGauges(IGaugeAdder.GaugeType.Arbitrum, minRelativeWeight, currentPeriod, singleArbGaugeBridgeETH);
+        _checkpointGauges(IGaugeAdder.GaugeType.Optimism, minRelativeWeight, currentPeriod, 0);
+        _checkpointGauges(IGaugeAdder.GaugeType.Gnosis, minRelativeWeight, currentPeriod, 0);
+        _checkpointGauges(IGaugeAdder.GaugeType.ZKSync, minRelativeWeight, currentPeriod, 0);
+
+        // transfer msg.value - total spent ETH back to sender.
+        // _getArbitrumBridgeCosts ensures the difference is always positive.
+        Address.sendValue(msg.sender, msg.value - totalArbBridgeETH);
+    }
+
+    /**
+     * @dev Returns a tuple with the ETH cost to checkpoint an individual gauge, and the cost to checkpoint all gauges.
+     * Reverts if the transaction does not have enough ETH to cover the total cost of the operation.
+     */
+    function _getArbitrumBridgeCosts() private view returns (uint256, uint256) {
+        uint256 totalGauges = _gauges[IGaugeAdder.GaugeType.Arbitrum].length();
+        if (totalGauges == 0) {
+            return (0, 0);
+        }
+        EnumerableSet.AddressSet storage arbitrumGauges = _gauges[IGaugeAdder.GaugeType.Arbitrum];
+
+        // We can do this cast safely because we check that the gauge was created from the ArbitrumRootGaugeFactory
+        // when we added it to the address set.
+        uint256 singleGaugeBridgeCost = ArbitrumRootGauge(arbitrumGauges.unchecked_at(0)).getTotalBridgeCost();
+        uint256 totalArbETH = singleGaugeBridgeCost * arbitrumGauges.length();
+        require(msg.value >= totalArbETH, "Not enough eth to cover arbitrum checkpoints");
+
+        return (singleGaugeBridgeCost, totalArbETH);
+    }
+
+    /**
+     * @dev Performs checkpoints for all gauges of the given type whose relative weight is at least the specified one.
+     * @param gaugeType - Type of the gauges to checkpoint.
+     * @param minRelativeWeight - Threshold to filter out gauges below it.
+     * @param currentPeriod - Current block time rounded down to the start of the week.
+     * @param costPerCheckpoint - Value in ETH to be spent for each gauge to checkpoint to cover bridging costs.
+     * This method doesn't check whether the caller transferred enough ETH to cover the whole operation.
+     */
+    function _checkpointGauges(
+        IGaugeAdder.GaugeType gaugeType,
+        uint256 minRelativeWeight,
+        uint256 currentPeriod,
+        uint256 costPerCheckpoint
+    ) private {
+        uint256 totalGauges = _gauges[gaugeType].length();
```

Comment:
Consider `totalTypeGauges` instead - the current name is slightly misleading.

Reply:
+1